lib/stream.js

```diff
@@ -84,9 +87,15 @@ function createReadableStream({ url, fetch, options = {} }) {
           break;
         }
 
-        controller.enqueue(
-          new ServerSentEvent(event.event, event.data, event.id)
-        );
+        let data = event.data;
+        if (
+          useFileOutput &&
+          typeof data === "string" &&
+          (data.startsWith("https:") || data.startsWith("data:"))
```

Comment:
I think we want to explicitly match that `data` consists of only a valid data uri. There is a reasonable chance that a language model might emit a `data:` line that starts with "data:" but less so that it will emit a line that consists only of a well formed data-uri.

🤔 That said, I did think today after reading this [post on other AI apis](https://til.simonwillison.net/llms/streaming-llm-apis) that we should move to structured outputs.

Perhaps the file stream should emit JSON.

```
data: {"type": "url", value: "data://..."}
```

And in future we refactor the text streaming interface to do the same:

```
data: {"type": "string", "data: and some more text"}
```